### PR TITLE
Refactor Cosmos repository to user internal model

### PR DESCRIPTION
**Pull request by jsancio**

### CODE REVIEW

**Comment by jsancio** [comment]
@tamarrow @cruhland, one more refactor. I should be ready to implement CosmosRepository and PackageCollection for the new internal model after this!

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/CosmosRepository.scala [hunk 1]
```diff
@@ -1,31 +1,62 @@
...
+
+final class DefaultCosmosRepository(
+    override val repository: rpc.v1.model.PackageRepository,
+    universeClient: V3UniverseClient
+)
+    extends V3CosmosRepository {
+
+  override def getPackageByReleaseVersion(
+      packageName: String,
+      releaseVersion: universe.v2.model.ReleaseVersion
```
Comment (cruhland) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/CosmosRepository.scala]: This should be `universe.v3.model.PackageDefinition.ReleaseVersion`.

**Comment by cruhland** [comment]
@jsancio just one comment, good to merge after it's addressed.

**Comment by tamarrow** [comment]
🎉

File: cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/CosmosRepository.scala (hunk 1 above)
Comment (jsancio) [inline cosmos-server/src/main/scala/com/mesosphere/cosmos/repository/CosmosRepository.scala]: Done.

**Comment by jsancio** [comment]
Thanks for the review. I will merge it when it is green.

**Pull request merged**